projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.spec.ts

```diff
@@ -29,7 +29,18 @@ class TestTextValueAsStringComponent {
 
   @Input() displayValue;
 }
+@Component({
+  selector: `kui-link-value`,
+  template: ``
+})
+class TestLinkValueComponent {
 
+  @Input() mode;
+
+  @Input() displayValue;
+  @Input() parentResource;
+  @Input() propType;
```

Comment:
@tobiasschweizer I have added the property type as an input parameter for the create mode. In displayEdit mode, it is set to displayValue.property. However, for the creation mode, I needed to know the property type. Is there any better way for this?

Reply:
Sorry, I am not sure I understand correctly. Couldn't you use the same Input prop for both cases, edit and create?

Reply:
@propType is an input parameter for both display/edit and create mode. 
If the mode === display or update: 
then propType=displayValue.property
in the create mode, proptype parameter should be specified as a parameter. See the template of component for both cases.

Reply:
so `propType` is the IRI of the property and there is another Input for the parent resource holding the entity defs?

Reply:
yes

Reply:
so there needs to be some logic that gets the prop definition from the entities

Reply:
if you want, I can have a look at it and see how the logic could look like

Reply:
@tobiasschweizer well, I do not see any problem with the current implementation. it works for both display/edit and the create mode.

Reply:
Sorry, I missed this at first. It is now clear to me.